torch/utils/module_tracker.py

```diff
@@ -49,17 +49,30 @@ def my_linear(m1, m2, bias):
     A Set containing the fqn for each module currently running their forward
     """
 
-    is_bw: bool
-    """
-    A boolean marking if this is currently running during the backward pass or not
-    """
-
     def __init__(self):
         self.parents = {"Global"}
-        # This is used to reset parents at the end of the backward
-        self.is_bw = False
         self._known_modules: weakref.WeakKeyDictionary = weakref.WeakKeyDictionary()
         self._seen_modules = set()
+        self._has_callback = False
+
+    def _maybe_set_engine_callback(self):
+        # This assumes no concurrent calls to backward
+        if self._has_callback:
+            return
+
+        def callback():
+            self.parents = {"Global"}
+            self._has_callback = False
+
+        torch.autograd.Variable._execution_engine.queue_callback(callback)
+        self._has_callback = True
+
+    @property
+    def is_bw(self):
+        """
+        A boolean marking if this is currently running during the backward pass or not
+        """
+        return torch._C._current_graph_task_id() != -1
```

Comment:
We don't have to pass is_bw to _get_append_fn and _get_pop_fn anymore right?

Reply:
I didn't make that change because:
- It is only used for error checking
- I do want this behavior when AC is involved for example, where the recompute is a forward hook happening during the backward